esp-wifi/src/wifi/mod.rs

```diff
@@ -872,14 +931,21 @@ impl TxToken for WifiTxToken {
 pub fn send_data_if_needed() {
     critical_section::with(|cs| {
         let mut queue = DATA_QUEUE_TX.borrow_ref_mut(cs);
+        let is_ap = *(IS_AP.borrow_ref(cs));
```

Comment:
Instead of global state, can we just call `esp_wifi_get_mode` here?

Reply:
That's much better - I think I wanted to avoid the FFI call but I doubt it will safe us anything plus this global state is quite ugly